sqlglot/generator.py

```diff
@@ -1130,7 +1135,15 @@ def expressions(self, expression, key=None, flat=False, indent=True, sep=", "):
         if flat:
             return sep.join(self.sql(e) for e in expressions)
 
-        expressions = self.sep(sep).join(self.sql(e) for e in expressions)
+        sql = [self.sql(e) for e in expressions]
+        # the only time leading_comma changes the output is if pretty print is enabled
+        # and there is more than 1 expression
+        if self._leading_comma and self.pretty and len(sql) > 1:
```

Comment:
is this len really necessary?

can you just do

"\n".join(f"{sep}s} if i > 0 else s for i, s in enumerate(sql))

Reply:
The `len()` is a O(1) operation, so I kept it in for the sake of clarity.  I personally find complex list comprehensions like the above hard read/reason about.  That being said, happy to make the change if you'd prefer that/it matches the aesthetic of the code base